### PR TITLE
feat: cfd-428 logic for the needs attention alert

### PR DESCRIPTION
**Pull request by JwanKhalaf**
# Description

-   Each service that has products where one or more of the products have a matching service id in the `servicesRequiringAttention` field in the `products` table, show a 'NEEDS ATTENTION' tag.

# Testing instructions

-   Manually add a service Id to the `servicesRequiringAttention` field on the `products` table and check to see if the tag appears for that service.

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-site/src/data/auroradb.ts [hunk 1]
```diff
@@ -1526,13 +1527,16 @@ export const getPointToPointProducts = async (nocCode: string): Promise<MyFaresP
...
         const queryInput = `      
-            SELECT id, lineId, matchingJsonLink, startDate, endDate
+            SELECT id, lineId, matchingJsonLink, startDate, endDate, servicesRequiringAttention 
             FROM products
             WHERE lineId <> ''
             AND nocCode = ?
         `;
 
-        return await executeQuery<MyFaresProduct[]>(queryInput, [nocCode]);
+        return (await executeQuery<RawMyFaresProduct[]>(queryInput, [nocCode])).map((row: any) => ({
```
[the remark this answers is not in the document]
Comment (JwanKhalaf) [reply repos/fdbt-site/src/data/auroradb.ts]: Sorted :+1:

**Review by Aaron-Collis-IW — APPROVED** [review]

File: repos/fdbt-site/src/pages/products/services.tsx [hunk 2]
```diff
@@ -88,14 +89,18 @@ export const getTag = (startDate: string, endDate: string | undefined): JSX.Elem
...
     } else if (startDateAsUnixTime > today) {
-        return <strong className="govuk-tag govuk-tag--blue">Pending</strong>;
+        return <strong className="govuk-tag govuk-tag--blue dft-table-tag">Pending</strong>;
     } else {
-        return <strong className="govuk-tag govuk-tag--red">Expired</strong>;
+        return <strong className="govuk-tag govuk-tag--red dft-table-tag">Expired</strong>;
     }
 };
 
+const getTheRequiresAttentionTag = (): JSX.Element => {
```
Comment (RobCatton) [inline repos/fdbt-site/src/pages/products/services.tsx:100]: can we just inline this?

Comment (JwanKhalaf) [reply repos/fdbt-site/src/pages/products/services.tsx]: Done :+1:

**Review by RobCatton — APPROVED** [review]

File: repos/fdbt-site/tests/pages/products/pointToPointProducts.test.tsx [hunk 3]
```diff
@@ -69,27 +69,31 @@ describe('myfares pages', () => {
                         matchingJsonLink: '/path/to/json',
                         startDate: '05/04/2018',
                         endDate: '10/04/2019',
+                        servicesRequiringAttention: [],
```
[the remark this answers is not in the document]
Comment (JwanKhalaf) [reply repos/fdbt-site/tests/pages/products/pointToPointProducts.test.tsx:73]: Added a test for `products/services.test.tsx`

**Pull request merged**